core/src/main/scala/kafka/security/auth/Authorizer.scala

```diff
@@ -106,16 +112,18 @@ trait Authorizer extends Configurable {
    *
    * {code}
    * // The following will get all ACLs from the literal resource path 'foo', which will only affect the topic named 'foo':
-   * authorizer.removeAcls(Resource(Topic, "foo", Literal))
+   * authorizer.removeAcls(Resource(Topic, "foo", LITERAL))
    *
    * // The following will get all ACLs from the special literal topic resource path '*', which affects all topics:
-   * authorizer.removeAcls(Resource(Topic, "*", Literal))
+   * authorizer.removeAcls(Resource(Topic, "*", LITERAL))
    *
    * // The following will get all ACLs from the prefixed resource path 'foo', which affects all topics whose name begins with 'foo':
-   * authorizer.removeAcls(Resource(Topic, "foo", Prefixed))
+   * authorizer.removeAcls(Resource(Topic, "foo", PREFIXED))
    * {code}
    *
    * @param resource the resource path to which the acls belong.
+   *                 the supplied resource will have a specific resource name type,
+   *                 i.e. the resource name type will not be ``ResourceNameType.ANY`` or ``ResourceNameType.UNKNOWN``.
```

Comment:
Is this true? I thought the current implementation allows you to pass in ResourceNameType.ANY to retrieve ACLs specified on either LITERAL or PREFIX?

Reply:
I believe this is true.  SimpleAclAuthorizer expects a concrete Resource, not something which could match multiple resources.  The code is here:

```
  override def removeAcls(resource: Resource): Boolean = {
    inWriteLock(lock) {
      val result = zkClient.deleteResource(resource)
      updateCache(resource, VersionedAcls(Set(), 0))
      updateAclChangedFlag(resource)
      result
    }
  }
```

The messiness here is that this constraint is not expressed in the type system.  We use the same enum to describe the name types a concrete resource can have vs. a resource filter.

Reply:
That's right. The enum has ANY and UNKNOWN values, but the code won't allow them to be passed to the authorizer.  I documented so that people implementing their own would know they aren't going to get these values.  It's part of the downside to standardizing on the Java ResourceNameType.